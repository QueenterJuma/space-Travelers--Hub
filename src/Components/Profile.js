import React from 'react';
import { useSelector } from 'react-redux';
import '../style/Mission.css';

const Profile = () => {
  const { missions } = useSelector((state) => state.missions);
  // const { rockets } = useSelector((state) => state.rockets);
  return (
    <div>
      <div className="mission-list">
        <h2>My Mission</h2>
        <table className="profile-table">
          <tbody>
            {missions
              .filter((mission) => mission.join)
              .map((join) => (
                <tr key={join.id}>
                  <td>{join.name}</td>
                </tr>
              ))}
          </tbody>
        </table>
      </div>
      {/* <div className="rocket-list">
        <h2>My Rockets</h2>
        <table className="profile-table">
          <tbody>
            {rockets
              .filter((rocket) => rocket.reserved)
              .map((reserved) => (
                <tr key={reserved.id}>
                  <td>{reserved.name}</td>
                </tr>
              ))}
          </tbody>
        </table>
      </div> */}
    </div>
  );
};

export default Profile;
