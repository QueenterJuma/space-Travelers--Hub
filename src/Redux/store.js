import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './Missions/missionSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
  },
});

export default store;
