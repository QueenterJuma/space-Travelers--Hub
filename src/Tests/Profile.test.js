import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../Redux/store';
import Profile from '../Components/Profile';

describe('Test Profile Component', () => {
  it('should render Profile component', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
