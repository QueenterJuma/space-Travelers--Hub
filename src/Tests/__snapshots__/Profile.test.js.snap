// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Profile Component should render Profile component 1`] = `
<div
  className="profile-container"
>
  <div
    className="mission-list"
  >
    <h2>
      My Mission
    </h2>
    <table
      className="profile-table"
    >
      <tbody />
    </table>
  </div>
  <div
    className="rockets-list"
  >
    <h2>
      My Rockets
    </h2>
    <table>
      <tbody />
    </table>
  </div>
</div>
`;
